### PR TITLE
Extension management - refactoring workspace trust signal

### LINKED ISSUE #116086: Install prompt after "restart to update"
Screenshot 2021-02-08 at 08 42 15

#### Shared process log

```
[2021-02-08 08:41:57.148] [sharedprocess] [info] Deleted from disk ms-vscode.vscode-typescript-next /Users/jrieken/.vscode-insiders/extensions/ms-vscode.vscode-typescript-next-4.2.20210203
[2021-02-08 08:41:59.156] [sharedprocess] [info] Installing extension: github.codespaces
[2021-02-08 08:41:59.158] [sharedprocess] [info] Installing extension: ms-vscode-remote.remote-ssh-edit-nightly
[2021-02-08 08:41:59.160] [sharedprocess] [info] Installing extension: ms-vscode-remote.remote-ssh-nightly
[2021-02-08 08:41:59.164] [sharedprocess] [info] Installing extension: ms-vscode.vscode-typescript-next
[2021-02-08 08:41:59.166] [sharedprocess] [info] Installing extension: ms-vscode.vscode-js-profile-flame
[2021-02-08 08:41:59.583] [sharedprocess] [info] Downloaded extension: ms-vscode-remote.remote-ssh-edit-nightly undefined
[2021-02-08 08:41:59.616] [sharedprocess] [info] Downloaded extension: ms-vscode.vscode-js-profile-flame undefined
[2021-02-08 08:41:59.660] [sharedprocess] [info] Downloaded extension: ms-vscode-remote.remote-ssh-nightly undefined
[2021-02-08 08:41:59.672] [sharedprocess] [info] Extracted extension to /Users/jrieken/.vscode-insiders/extensions/.ms-vscode-remote.remote-ssh-edit-nightly-2021.2.6660: ms-vscode-remote.remote-ssh-edit-nightly
[2021-02-08 08:41:59.672] [sharedprocess] [info] Renamed to /Users/jrieken/.vscode-insiders/extensions/ms-vscode-remote.remote-ssh-edit-nightly-2021.2.6660
[2021-02-08 08:41:59.675] [sharedprocess] [info] Installation completed. ms-vscode-remote.remote-ssh-edit-nightly
[2021-02-08 08:41:59.706] [sharedprocess] [info] Downloaded extension: github.codespaces undefined
[2021-02-08 08:41:59.758] [sharedprocess] [info] Extracted extension to /Users/jrieken/.vscode-insiders/extensions/.ms-vscode.vscode-js-profile-flame-0.0.14: ms-vscode.vscode-js-profile-flame
[2021-02-08 08:41:59.761] [sharedprocess] [info] Renamed to /Users/jrieken/.vscode-insiders/extensions/ms-vscode.vscode-js-profile-flame-0.0.14
[2021-02-08 08:41:59.785] [sharedprocess] [info] Installation completed. ms-vscode.vscode-js-profile-flame
[2021-02-08 08:41:59.785] [sharedprocess] [info] Extensions installed successfully: ms-vscode-remote.remote-ssh-edit-nightly
[2021-02-08 08:41:59.874] [sharedprocess] [info] Extensions installed successfully: ms-vscode.vscode-js-profile-flame
[2021-02-08 08:42:00.114] [sharedprocess] [info] Installing extension: ms-vscode.vscode-js-profile-flame
[2021-02-08 08:42:00.119] [sharedprocess] [info] Installing extension: ms-vscode-remote.remote-ssh-edit-nightly
[2021-02-08 08:42:00.169] [sharedprocess] [info] Downloaded extension: ms-vscode.vscode-typescript-next undefined
[2021-02-08 08:42:00.211] [sharedprocess] [info] Downloaded extension: ms-vscode.vscode-js-profile-flame undefined
[2021-02-08 08:42:00.260] [sharedprocess] [info] Downloaded extension: ms-vscode-remote.remote-ssh-edit-nightly undefined
[2021-02-08 08:42:00.306] [sharedprocess] [info] Extracted extension to /Users/jrieken/.vscode-insiders/extensions/.ms-vscode-remote.remote-ssh-nightly-2021.2.6660: ms-vscode-remote.remote-ssh-nightly
[2021-02-08 08:42:00.306] [sharedprocess] [info] Renamed to /Users/jrieken/.vscode-insiders/extensions/ms-vscode-remote.remote-ssh-nightly-2021.2.6660
[2021-02-08 08:42:00.309] [sharedprocess] [info] Installation completed. ms-vscode-remote.remote-ssh-nightly
[2021-02-08 08:42:00.532] [sharedprocess] [info] Extensions installed successfully: ms-vscode-remote.remote-ssh-nightly
[2021-02-08 08:42:00.601] [sharedprocess] [info] Extracted extension to /Users/jrieken/.vscode-insiders/extensions/.ms-vscode-remote.remote-ssh-edit-nightly-2021.2.6660: ms-vscode-remote.remote-ssh-edit-nightly
[2021-02-08 08:42:00.602] [sharedprocess] [info] Renamed to /Users/jrieken/.vscode-insiders/extensions/ms-vscode-remote.remote-ssh-edit-nightly-2021.2.6660
[2021-02-08 08:42:00.606] [sharedprocess] [info] Installation completed. ms-vscode-remote.remote-ssh-edit-nightly
[2021-02-08 08:42:00.642] [sharedprocess] [info] Extensions installed successfully: ms-vscode-remote.remote-ssh-edit-nightly
[2021-02-08 08:42:00.656] [sharedprocess] [info] Extracted extension to /Users/jrieken/.vscode-insiders/extensions/.ms-vscode.vscode-js-profile-flame-0.0.14: ms-vscode.vscode-js-profile-flame
[2021-02-08 08:42:00.656] [sharedprocess] [info] Renamed to /Users/jrieken/.vscode-insiders/extensions/ms-vscode.vscode-js-profile-flame-0.0.14
[2021-02-08 08:42:00.659] [sharedprocess] [info] Installation completed. ms-vscode.vscode-js-profile-flame
[2021-02-08 08:42:00.770] [sharedprocess] [info] Extensions installed successfully: ms-vscode.vscode-js-profile-flame
[2021-02-08 08:42:01.124] [sharedprocess] [info] Extracted extension to /Users/jrieken/.vscode-insiders/extensions/.github.codespaces-0.7.1: github.codespaces
[2021-02-08 08:42:01.124] [sharedprocess] [info] Renamed to /Users/jrieken/.vscode-insiders/extensions/github.codespaces-0.7.1
[2021-02-08 08:42:01.128] [sharedprocess] [info] Installation completed. github.codespaces
[2021-02-08 08:42:01.201] [sharedprocess] [info] Extensions installed successfully: github.codespaces
[2021-02-08 08:42:01.468] [sharedprocess] [info] Extracted extension to /Users/jrieken/.vscode-insiders/extensions/.ms-vscode.vscode-typescript-next-4.2.20210207: ms-vscode.vscode-typescript-next
[2021-02-08 08:42:01.468] [sharedprocess] [info] Renamed to /Users/jrieken/.vscode-insiders/extensions/ms-vscode.vscode-typescript-next-4.2.20210207
[2021-02-08 08:42:01.470] [sharedprocess] [info] Installation completed. ms-vscode.vscode-typescript-next
[2021-02-08 08:42:01.527] [sharedprocess] [info] Extensions installed successfully: ms-vscode.vscode-typescript-next
[2021-02-08 08:42:36.498] [sharedprocess] [info] Starting to clean up unused language packs.

```

from another window I see that the extension is there

Screenshot 2021-02-08 at 09 20 19

### PR DESCRIPTION
**Pull request by lszomoru**
This PR fixes #116086

### CODE REVIEW

File: src/vs/workbench/services/extensionManagement/common/extensionManagementService.ts [hunk 1]
```diff
@@ -269,7 +279,9 @@ export class ExtensionManagementService extends Disposable implements IWorkbench
 					servers.push(this.extensionManagementServerService.localExtensionManagementServer);
 				}
 			}
-			return Promises.settled(servers.map(server => server.extensionManagementService.installFromGallery(gallery, installOptions))).then(([local]) => local);
+			return this.promptForTrustIfNeededAndInstall(manifest, () => {
+				return Promises.settled(servers.map(server => server.extensionManagementService.installFromGallery(gallery, installOptions))).then(([local]) => local);
+			});
```
Comment (sandy081) [inline src/vs/workbench/services/extensionManagement/common/extensionManagementService.ts:284]: I would decouple `promptForTrustIfNeeded` and `install` operations for simplification, say:

```
await this.checkForTrustedWorkspace();
return Promises.settled(servers.map(server => server.extensionManagementService.installFromGallery(gallery, installOptions))).then(([local]) => local);
```

**Review by sandy081 — APPROVED** [review]
Other than suggested simplification, changes LGTM

File: src/vs/workbench/services/extensionManagement/common/extensionManagementService.ts (hunk 1 above)
Comment (lszomoru) [reply src/vs/workbench/services/extensionManagement/common/extensionManagementService.ts:284]: Fixed. Can you please have another look? Thanks!

**Review by sandy081 — APPROVED** [review]
Thanks for changes. LGTM

**Pull request merged**